r2-core/src/test/java/test/r2/transport/http/client/TestAsyncPool.java

```diff
@@ -647,34 +648,35 @@ public void testObjectsAreNotCreatedWhenThereAreNoWaiters(int numberOfCheckouts,
     }
     catch (Exception e)
     {
-      Assert.fail("Unexpected interruption", e);
+      Assert.fail("Did not complete unblocked object creations on time, Unexpected interruption", e);
     }
 
     // Making sure the rate limiter pending tasks are submitted to the executor
-    while (rateLimiter.numberOfPendingTasks() > 0)
-    {
-      try
-      {
-        Thread.sleep(1);
-      }
-      catch (Exception e)
+    AssertionMethods.assertWithTimeout(5000, () -> {
+      while (rateLimiter.numberOfPendingTasks() > 0)
       {
-        Assert.fail("Unexpected interruption", e);
-      }
-    }
```

Comment:
AssertionMethods.assertWithTimeout(5000, () ->       Assert.assertEquals(rateLimiter.numberOfPendingTasks(),0,"Number of tasks has to drop to 0")

Reply:
Same for others

Reply:
Neat !